functions

```diff
@@ -152,3 +152,17 @@ service_url() {
   local SERVICE_DNS_HOSTNAME="$(service_dns_hostname "$SERVICE")"
   echo "$PLUGIN_SCHEME://postgres:$PASSWORD@$SERVICE_DNS_HOSTNAME:${PLUGIN_DATASTORE_PORTS[0]}/$DATABASE_NAME"
 }
+
+service_external_url() {
+  local SERVICE="$1"
+  local SERVICE_ROOT="$PLUGIN_DATA_ROOT/$SERVICE"
+  local PORT_FILE="$SERVICE_ROOT/PORT"
+  [[ ! -f $PORT_FILE ]] && echo '-' && return 0
+
+  local PORTS=($(cat "$PORT_FILE"))
```

Comment:
This port may not be exposed to the public, so this external dsn would fail to work.

Reply:
The port is _by default_ only on the internal ip address. You'd need to call `SERVICE:expose` on the service (with the port you want to map it to specified) and only then would this code work.

Reply:
Yes, maybe I missed some explanation. I currently run `ssh dokku@dokku.me postgres:expose postgres-database` before `pgcli $(ssh dokku@dokku.me postgres:info postgres-database --external-dsn)`. If not `External dsn` returns `-` as `Exposed ports` do.

```
$ ssh dokku@dokku.me postgres:info postgres-database          
=====> Container Information
       Config dir:          /var/lib/dokku/services/postgres/postgres-database/config
       Data dir:            /var/lib/dokku/services/postgres/postgres-database/data
       Dsn:                 postgres://postgres:xxxxxxxxxxxx@dokku-postgres-postgres-database:5432/postgres_database
       External dsn:        -                        
       Exposed ports:       -                        
       Id:                  9dc918204b5fb0a4b8848a71
       Internal ip:         172.17.0.3               
       Links:               postgres-database       
       Service root:        /var/lib/dokku/services/postgres/postgres-database
       Status:              running                  
       Version:             postgres:10.4  
$ ssh dokku@dokku.me postgres:expose postgres-database
$ ssh dokku@dokku.me postgres:info postgres-database          
=====> Container Information
       Config dir:          /var/lib/dokku/services/postgres/postgres-database/config
       Data dir:            /var/lib/dokku/services/postgres/postgres-database/data
       Dsn:                 postgres://postgres:xxxxxxxxxxxx@dokku-postgres-postgres-database:5432/postgres_database
       External dsn:        postgres://postgres:xxxxxxxxxxxx@dokku.me:28804/postgres_database                        
       Exposed ports:       5432->28804                        
       Id:                  9dc918204b5fb0a4b8848a71
       Internal ip:         172.17.0.3               
       Links:               postgres-database       
       Service root:        /var/lib/dokku/services/postgres/postgres-database
       Status:              running                  
       Version:             postgres:10.4 
```

Maybe I have to update the README for more explanation.

Reply:
Right I'm just concerned that if the db is not exposed, then people will get confused and start filing issues about how this doesn't work.

Reply:
You're right, I'll work on updating README and maybe add precision on command help info about the option.

Reply:
I do some changes on README and flag description.

Reply:
Should be empty if the service is not exposed.

Reply:
`External dsn:` already returns `-` if the service is not exposed.

Reply:
Does this fine for you?